src/main/java/org/opentestsystem/delivery/logging/EventInfo.java

```diff
@@ -1,79 +1,93 @@
  */
-public abstract class EventInfo {
+public class EventInfo {
+  private String event;
+  private String checkpoint;
+  private String message;
+  private final Map<EventData, Object> data = new HashMap<>();
 
   /**
    * @return Name of the current event
    */
-  abstract String event();
+  public String event() {
+    return event;
+  }
+
+  void setEvent(final String event) {
+    this.event = event;
+  }
 
   /**
    * @return Moment in time of the event, such as the entry, exit or error
    */
-  abstract Optional<String> checkpoint();
+  public Optional<String> checkpoint() {
+    return Optional.fromNullable(checkpoint);
+  }
+
+  void setCheckpoint(final String checkpoint) {
+    this.checkpoint = checkpoint;
+  }
 
   /**
    * @return Optional message for the event
    */
-  abstract Optional<String> message();
+  public Optional<String> message() {
+    return Optional.fromNullable(message);
+  }
+
+  void setMessage(final String message) {
+    this.message = message;
+  }
 
   /**
    * @return Event data payload that is used to access the event from log files
    */
-  abstract Map<EventData, Object> data();
-
-  public static EventInfo create(String event, String checkpoint, String message, Map<EventData, Object> data) {
-    return builder()
-      .event(event)
-      .checkpoint(Optional.of(checkpoint))
-      .message(Optional.of(message))
-      .data(data)
-      .build();
+  public Map<EventData, Object> data() {
+    return data;
   }
 
-  public static EventInfo create(String event, String checkpoint, Map<EventData, Object> data) {
-    return builder()
-      .event(event)
-      .checkpoint(Optional.of(checkpoint))
-      .message(Optional.<String>absent())
-      .data(data)
-      .build();
+  void setData(final Map<EventData, Object> data) {
+    this.data.putAll(data);
   }
 
-  public static EventInfo create(String event, Map<EventData, Object> data) {
-    return builder()
-      .event(event)
-      .checkpoint(Optional.<String>absent())
-      .message(Optional.<String>absent())
-      .data(data)
-      .build();
+  /**
+   * Create a builder containing this EventInfo's values.
+   *
+   * @return A builder
+   */
+  public EventInfoBuilder toBuilder() {
+    return builder().copy(this);
   }
 
-  public static Builder builder() {
-    return new AutoValue_EventInfo.Builder();
+  /**
+   * Create a new empty builder.
+   *
+   * @return A builder
+   */
+  public static EventInfoBuilder builder() {
+    return new EventInfoBuilder();
   }
 
-  abstract Builder toBuilder();
-
   /**
    * Copy of this EventInfo with checkpoint changed
    *
    * @param checkpoint new value of checkpoint
    * @return Copy of this EventInfo with a new checkpoint
    */
-  public EventInfo withCheckpoint(String checkpoint) {
-    return toBuilder().checkpoint(Optional.of(checkpoint)).build();
+  public EventInfo withCheckpoint(final String checkpoint) {
+    return toBuilder().checkpoint(checkpoint).build();
   }
```

Comment:
Personal preference, but I'm not a fan of convenience methods on models.  I like my models to be simple POJOs.  But since it was existing functionality I left it in here.